modules/terrain/test/terrain-loader.spec.js

```diff
@@ -0,0 +1,119 @@
+/* eslint-disable max-len */
+import test from 'tape-promise/tape';
+import {validateLoader, validatePointCloudCategoryData} from 'test/common/conformance';
+
+import {TerrainLoader, TerrainWorkerLoader} from '@loaders.gl/terrain';
+import {setLoaderOptions, load} from '@loaders.gl/core';
+
+const MAPBOX_TERRAIN_PNG_URL = '@loaders.gl/terrain/test/data/mapbox.png';
+const TERRARIUM_TERRAIN_PNG_URL = '@loaders.gl/terrain/test/data/terrarium.png';
+
+setLoaderOptions({
+  terrain: {
+    workerUrl: 'modules/terrain/dist/terrain-loader.worker.js'
+  }
+});
+
+test('TerrainLoader#loader objects', async t => {
+  validateLoader(t, TerrainLoader, 'TerrainLoader');
+  validateLoader(t, TerrainWorkerLoader, 'TerrainWorkerLoader');
+  t.end();
+});
+
+test('TerrainLoader#parse mapbox', async t => {
+  const options = {
+    terrain: {
+      elevationDecoder: {
+        rScaler: 65536 * 0.1,
+        gScaler: 256 * 0.1,
+        bScaler: 0.1,
+        offset: -10000
+      },
+      meshMaxError: 5.0,
+      bounds: [83, 329.5, 83.125, 329.625] // note: not the real tile bounds
+    }
+  };
+  const data = await load(MAPBOX_TERRAIN_PNG_URL, TerrainLoader, options);
+  validatePointCloudCategoryData(t, data); // TODO: should there be a validateMeshCategoryData?
```

Comment:
@Pessimistress @ibgreen I was wondering if we should make a `validateMeshCategoryData`. I saw that `validatePointCloudCategoryData` was used for other mesh types, and seems to be doing helpful validation here too. Another function may not be needed, but wanted to check.

Reply:
We no longer have a point cloud category. This should really be called `validateMeshCategoryData`.